### PR TITLE
Connect twitter account

### PR DESCRIPTION
**Pull request by sedubois**
- [x] connect twitter account
- [x] thank backers after each donation
- [x] monthly thank you tweet to all backers
- [x] configurability to opt-in/out of both tweet types
- [x] read templates from group.settings when present
- [ ] form to opt-in/opt-out of both tweet types
- [ ] form to edit templates stored in group.settings

### CODE REVIEW

**Comment by asood123** [comment]
just like github integration, for now let's not put this behind a user login. To try it out quickly, we can put it at an unlinked url. Like opencollective.com/:groupslug/connect/twitter and get the users to oauth from that screen. Otherwise, this will be blocked until we have proper login.

**Comment by sedubois** [comment]
@asood123 I removed need for user login

**Comment by sedubois** [comment]
Ready for review/merge

**Comment by sedubois** [comment]
Need to change scheduler to call `npm run report:weekly` and `npm run report:monthly` when deploying this PR.

**Comment by sedubois** [comment]
Added some documentation in wiki: https://github.com/OpenCollective/OpenCollective/wiki/Twitter

File: server/lib/twitter.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function getDefaultTemplate(templateType, singular) {
+  switch (templateType) {
+    case activityType.GROUP_TRANSACTION_CREATED:
+      return `#1 thanks for backing us!`;
+
+    case activityType.GROUP_MONTHLY:
+      if (singular) {
+        return `Thank you #2for supporting our collective`;
```
Comment (asood123) [inline server/lib/twitter.js]: Should there be a space after "#2"?

File: server/lib/twitter.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      } else {
+        return `Thanks to our #1 backers and sponsors #2for supporting our collective`;
```
Comment (asood123) [inline server/lib/twitter.js]: Same comment here.

**Comment by asood123** [comment]
Besides that spacing comment, everything else looks good to merge.

File: server/lib/twitter.js (hunk 2 above)
Comment (asood123) [inline server/lib/twitter.js]: Nevermind. I see that you are adding a space above in the script.

File: server/lib/twitter.js (hunk 1 above)
Comment (sedubois) [inline server/lib/twitter.js]: No, so that it doesn't show two consecutive spaces if list of twitter handles is empty

**Pull request merged**